module/VuFind/src/VuFind/RecordDriver/SolrDefaultWithoutSearchServiceFactory.php

```diff
@@ -67,8 +67,15 @@ public function __invoke(
         if (!empty($options)) {
             throw new \Exception('Unexpected options sent to factory.');
         }
-        $config = $container->get(\VuFind\Config\PluginManager::class)
-            ->get('searches');
+
+        if (str_contains($requestedName, 'Search2')) {
```

Comment:
What is the purpose of this special case? This feels like logic that is not in the most appropriate place. Could we instead create a custom Search2Factory to accomplish the same goal in a more targeted way?

Reply:
I did not want to duplicate the code. But you are probably right that this is not the best way to do it. I created a Search2DefaultFactory.